### ISSUE TITLE
Is the dropdown component really accessible?

### ISSUE THREAD

**Issue by brunano21** [ng-bootstrap/ng-bootstrap#1998]
Hi,
I have looked into the drop-down implementation and it seems it does not follow any of the ARIA 1.0 or ARIA 1.1, in order to be an accessible widget. No role attribute, not items navigation through up/down keys and so forth.
Thus I wonder if this is intentional or something still in progress.

Plunker: http://plnkr.co/edit/HdaNtCIfXcLbB2VqCSl5?p=preview

Angular:
4.0.3
ng-bootstrap:
v1.0.0-beta.5
Bootstrap:
4.0.0-beta.2

**Comment by pkozlowski-opensource** [comment]
> No role attribute, 

Could you be more specific here? Which role an on which element do you expect, exactly?

> not items navigation through up/down keys and so forth.

This is work in progress.

**Comment by brunano21** [comment]
Sorry for the late reply
> Could you be more specific here? Which role an on which element do you expect, exactly?

I would say one between [`menu`](https://www.w3.org/TR/wai-aria-1.1/#menu) or [`listbox`](https://www.w3.org/TR/wai-aria-1.1/#listbox) on the same node where `aria-haspopup` and `aria-expanded` attributes are.

**Comment by AntonLebiodkin** [comment]
Could arrow-navigation and trap focus be added to dropdowns?
I think the PR for it is already exists here:
https://github.com/ng-bootstrap/ng-bootstrap/pull/1417

**Comment by tomwanzek** [comment]
Thanks for the hard work on the project.

Is there any progress on getting native Keyboard support for the dropdown? It's unfortunate to have to wrap the drop-down and implement custom keyboard event handling.

@pkozlowski-opensource  Is there any reason, why PR #1417 cannot be finalized and published with a minor release?

**Comment by maxokorokov** [comment]
Closed in #2683

**Comment by divyanshumehta** [comment]
hey @pkozlowski-opensource  and @maxjoehnk   agreed with @brunano21 . The presence of both `aria-haspopup` and `aria-expanded` makes it impossible for screen reader likes NVDA/JAWS to read the dropdown. Removing the `aria-haspopup` attribute does the trick. The link below shows how the `aria-haspopup` label should only be used for something that covers the main page. Removing `aria-haspopup` can make the dropdown component really accessible. 

https://www.levelaccess.com/expanding-the-use-of-the-aria-haspopup-property/

**Comment by gilluminate** [comment]
For anyone coming here with questions, the suggestion to remove `aria-haspopup` by @divyanshumehta is entirely the wrong solution. Removing that will break the experience for screen reader users. The article referenced actually says the *exact opposite* of what he is suggesting.
